src/main/webapp/script.js

```diff
@@ -18,3 +18,48 @@ function createElementByTag(text, tag) {
   element.innerText = text;
   return element;
 }
+
+/**
+ * parse.js
+ *
+ * MediaWiki API Demos
+ * Demo of `Parse` module: Parse content of a page
+ *
+ * MIT License
+ */
+
+async function parseData() {
+    // Parse data from single page.
+    const url = "https://en.wikibooks.org/w/api.php?" +
+    new URLSearchParams({
+        origin: "*",
+        action: "parse",
+        page: "Cookbook:Racuszki",
+        format: "json",
+        prop: "text",
+    });
+
+    const request = await fetch(url);
+    const json = await request.json();
+    const container = document.getElementById("data-container");
+    container.innerHTML = json.parse.text['*'];
+
+    const ingredients = [];
+    const ingrHTML = document.getElementsByTagName("ul").item(0).children;
+    console.log(ingrHTML);
+    Array.from(ingrHTML).forEach((el) => {
+        ingredients.push(el.innerText);
+        });
+    
+    // Create JSON String for Meal object.
+    const data = JSON.stringify(
+        {
+            id: 25,
+            title: "racuszki",
+            description: "Racuszki are pancakes made from batter fried on a hot griddle or a pan. They are a traditional Polish meal.",
+            type: "Pancake",
+            ingredients: ingredients
+        });
+     
+    window.location.replace("/import?input="+encodeURIComponent(data));
```

Comment:
Please use `fetch` instead of `window.location.replace`. This will make things easier, especially if you'll want to process multiple Wikipedia articles at a time.
Also, `fetch` has a convenient `body:` parameter where you can put the JSON you want to send.